### PR TITLE
feat: add rate limit per minute

### PR DESCRIPTION
**Pull request by missuo**
There are no conflicts anymore.

### CODE REVIEW

**Comment by funnyzak** [comment]
1. You need to add the variable `RATE_LIMIT` to the template configuration file **config.env.example**.

2. Need to add the variable `RATE_LIMIT` to **READEME.md** and **README_CN.md**.

File: main.go [hunk 1]
```diff
@@ -307,11 +327,16 @@ func main() {
...
+	if config.ConfigInstance.RateLimit == 0 {
+		rateLimiterEnabled = false
+	}
+
 	router := gin.Default()
 	router.Use(CORSMiddleware())
 	router.Use(LoggerHandler())
 
-	router.POST("/v1/chat/completions", chatCompletions)
+	router.POST("/v1/chat/completions", RateLimiterHandler(rateLimiterEnabled, config.ConfigInstance.RateLimit), chatCompletions)
```
Comment (funnyzak) [inline main.go:339]: I think the variable `rate Limiter Enabled` can be removed? Write the enable logic in the retelimit handler.

File: README.md [hunk 2]
```diff
@@ -68,6 +68,7 @@ DEBUG=false # Whether to enable debug mode, more logs will be output after enabl
 LOGGING=true # Whether to enable logging, default is true.
 LOG_LEVEL=info # Log level, optional values: panic, fatal, error, warn, info, debug, trace (Note: effective only when LOGGING=true), default is info.
 COPILOT_TOKEN=ghp_xxxxxxx # Default Github Copilot Token, if this item is set, the Token carried with the request will be ignored. Default is empty.
+RATE_LIMIT= # The number of requests allowed per minute, if this item is not set, means there is no limit.
```
Comment (funnyzak) [inline README.md:71]: Isn’t it better to express it like this?

> The number of requests allowed per minute, if 0 there is no limit, default is 0.

Comment (funnyzak) [reply README.md:71]: RATE_LIMIT=0 # The number of requests allowed per minute, if 0 there is no limit, default is 0.

Comment (missuo) [reply README.md]: OK!

**Review by funnyzak — APPROVED** [review]

**Review by aaamoon — APPROVED** [review]

**Comment by aaamoon** [comment]
@funnyzak @Geniucker Whether this PR can be merged or not, I have tested it.

**Comment by funnyzak** [comment]
I've tested it and it's fine.

**Pull request merged**